go/vt/schemadiff/schema_diff.go

```diff
@@ -346,23 +344,23 @@ func (d *SchemaDiff) OrderedDiffs(ctx context.Context) ([]EntityDiff, error) {
 	return orderedDiffs, nil
 }
 
-// CapableOfInstantDDL returns `true` if all diffs are capable of instant DDL, or are otherwise trivially
-// instantaneously applicable (such as `CREATE TABLE` or `ALTER VIEW`). The answer essentially indicates whether
-// the entire set of changes can be applied as an immediate operation.
-func (d *SchemaDiff) CapableOfInstantDDL(ctx context.Context, capableOf capabilities.CapableOf) (bool, error) {
-	if capableOf == nil {
-		return false, nil
-	}
-	var errs error
-	allCapable := true
+// InstantDDLCapability returns an overall summary of the ability of the diffs to run with ALGORITHM=INSTANT.
+// It is a convenience method, whose logic anyone can reimplement.
+func (d *SchemaDiff) InstantDDLCapability() InstantDDLCapability {
+	// The general logic: we return "InstantDDLCapabilityPossible" if there is one or more diffs that is capable of
+	// ALGORITHM=INSTANT, and zero or more diffs that are irrelevant.
```

Comment:
> Would it be simpler and more clear to say that we return "InstantDDLCapabilityPossible" if there are no diffs which are impossible to run with ALGORITHM=INSTANT?

The distinction we're trying to make is between "there's nothing relevant to INSTANT" and "there's nothing to prevent an INSTANT". 

Example to the first would be a change with only a bunch of `CREATE` and `DROP` statements. None of them are "impossible" for `INSTANT`, but they're all just irrelevant. For this diff, you can technically say it's eligigle for INSTANT, but that's also misleading. There would be nothing to do. You cannot improve, change, or hasten the migrations in any way.

Example to the second is a bunch of `CREATE` and `DROP`, and one `ALTER TABLE ... ADD COLUMN`. Here, as a whole, we say that it's _possible_ to run INSTANT DDL. There is an actual different course of action to the diff as a whole, if we apply ALGORITHM=INSTANT.

The distinction is really important, because we want to be able to look at a diff and say "hey, if you want this diff to run faster, you _could_ add ALGORITHM=INSTANT". This kind of suggestion would be misleading in the 1st example, and beneficial in the 2nd example.

Reply:
> But ALSO have one more more that are impossible. No? It's obviously minor.

Yes, clarified in code comment.